### PR TITLE
Add k8s 1.17.0 support to kubevirtci

### PR DESCRIPTION
**Pull request by dhiller**
Additionally cluster-provision/k8s/check-cluster-up.sh (for the moment
only used in 1.17) checks whether the cluster can come up with the
local image. Therefore added new parameter container-registry to only
download image if required.
Extracted the image hashes into cluster-up/cluster/images.sh that is
included in the k8s provider shell scripts.
Added new env var KUBEVIRTCI_GOCLI_CONTAINER to override the gocli
for local development.

**Note: new gocli image pushed and integrated.**

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@dhiller Can you create prow job for the script  `cluster-provision/k8s/check-cluster-up.sh`?

**Comment by qinqon** [comment]
@dhiller to push you run push-gcli job with "/test push-gocli"

File: cluster-up/cluster/images.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -e
+
+declare -A IMAGES
+IMAGES[k8s-1.16.2]="k8s-1.16.2@sha256:5bae6a5f3b996952c5ceb4ba12ac635146425909801df89d34a592f3d3502b0c"
+IMAGES[k8s-1.15.1]="k8s-1.15.1@sha256:14d7b1806f24e527167d2913deafd910ea46e69b830bf0b094dde35ba961b159"
+IMAGES[k8s-1.14.6]="k8s-1.14.6@sha256:ec29c07c94fce22f37a448cb85ca1fb9215d1854f52573316752d19a1c88bcb3"
+IMAGES[k8s-1.13.3]="k8s-1.13.3@sha256:afbdd9b4208e5ce2ec327f302c336cea3ed3c22488603eab63b92c3bfd36d6cd"
+IMAGES[k8s-1.11.0]="k8s-1.11.0@sha256:696ba7860fc635628e36713a2181ef72568d825f816911cf857b2555ea80a98a"
```
Comment (qinqon) [inline cluster-up/cluster/images.sh]: Can we add the gocli here too ? IMAGES[gocli]="...

File: cluster-up/cluster/ephemeral-provider-common.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@ set -e
 if [ "${KUBEVIRTCI_RUNTIME}" = "podman" ]; then
 	_cli="pack8s"
 else
```
Comment (qinqon) [inline cluster-up/cluster/ephemeral-provider-common.sh]: source images.sh here so it's not needed to be imported at provider.sh

File: cluster-up/cluster/k8s-1.11.0/provider.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-image="k8s-1.11.0@sha256:696ba7860fc635628e36713a2181ef72568d825f816911cf857b2555ea80a98a"
+source ${KUBEVIRTCI_PATH}/cluster/images.sh
```
Comment (qinqon) [inline cluster-up/cluster/k8s-1.11.0/provider.sh]: Let's do this at ephemeral-provider-common.sh

File: cluster-up/cluster/images.sh (hunk 1 above)
Comment (dhiller) [reply cluster-up/cluster/images.sh]: Oh, great idea :)

**Comment by dhiller** [comment]
@qinqon fixes applied, please review again.

**Pull request merged**